hips/utils/wcs.py

```diff
@@ -0,0 +1,86 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import numpy as np
+from astropy.wcs import WCS
+from astropy.coordinates import SkyCoord
+
+__all__ = [
+    'WCSGeometry',
+]
+
+
+class WCSGeometry:
+    """Container for WCS object and image shape.
+
+    Parameters
+    ----------
+    wcs : `~astropy.wcs.WCS`
+        WCS projection object
+    shape : tuple
+        Shape of the image
+    """
+
+    def __init__(self, wcs: WCS, shape: tuple) -> None:
+        self._wcs = wcs
+        self._shape = shape
+
+    @property
+    def shape(self) -> None:
```

Comment:
Suggest to remote the `property` for shape and wcs, and to instead store `self.wcs=wcs` and `self.shape` directly in `__init__`.